crates/sdk/macros/src/lib.rs

```diff
@@ -92,17 +92,13 @@ pub fn emit(input: TokenStream) -> TokenStream {
 pub fn bail(input: TokenStream) -> TokenStream {
     let input = parse_macro_input!(input as Expr);
 
-    if let Expr::Lit(lit) = &input {
-        if let syn::Lit::Str(str) = &lit.lit {
-            return quote! {
-                return Err(::calimero_sdk::types::Error::msg(#str));
-            }
-            .into();
-        }
-    }
+    quote! {{
+        let wrapped = ::calimero_sdk::types::__private::Wrap(#input);
 
-    quote! {
-        return Err(::calimero_sdk::types::Error::from(#input));
-    }
+        {
+            use ::calimero_sdk::types::__private::*;
+            return ::core::result::Result::Err(wrapped.into_error());
+        }
+    }}
```

Comment:
this was written this way to sanitize the environment, something [return type](https://github.com/calimero-network/core/blob/0588c5e4cc0f9db2a83d76e4ce4c4a942d813e40/crates/sdk/macros/src/logic/method.rs#L136-L150) generalization can borrow from